mne/time_frequency/psd.py

```diff
@@ -128,16 +138,26 @@ def psd_array_welch(x, sfreq, fmin=0, fmax=np.inf, n_fft=256, n_overlap=0,
                              for d in x_splits)
 
     # Combining, reducing windows and reshaping to original data shape
-    psds = np.concatenate([np.nanmean(f_s, axis=-1)
-                           for f_s in f_spectrogram], axis=0)
-    psds.shape = np.hstack([dshape, -1])
+    # XXX : we can certainly avoid the allocation before the mean
+    #       (this would require pushing reduction into fixes._spectrogram)
```

Comment:
I had changed it so the allocation is avoided, at least for `nanmean`. You should be able to keep it or do something similar.